.travis.yml

```diff
@@ -16,14 +17,16 @@ deploy:
   api_key:
     secure: MZOznxDJeaIT3IQYDZHoH7oazeuM8ekgZzvnUTq0gMHZc1gwB9uEYjvCwSVo0OlUKFPymarhv/65GolggV1+fZzig5HPOpJltTFj3w9RBXZZzZiNZ3X+eexyW+so07BtV1S+jytA/Dd8MpnT9EmCYTzpg6Gp/NosyQJ9FbS7hm8UQcH7PLiBWAeoL1Wb/aLZQZUQRBiYr8PmZ6lzVTY+w8h5KISi7TrL6EzbwXj3j3muCGm/9ZqX/6CAxmAydU0ozCjl4QksRqn2MDWgUSYgga2GKBqsTpU4hX05QCEXD/G2BcGmoaOFP29na+zhXlCZ9j7DrOCMisEWH/DiWwqWaNlqUgO81pGbQDayvpZAMGRrYsaSQTz7oEbuU88NeDFriy44WFGE+eE73ykPFQl6JNnptE3mF6dRVPb6GZhG0nzM+UKfNw4hhcWo45hmrfqLS8MJNci9O3HbMy+2Ka45u64RlX2AUnmNexdPFDmZSjARLrnFT4VUuTEeb3CRS8TmNaWEM1o2yWYgtCkM9qnhiAxlrcDt1Y/bqhgbhzoheslHmi+n9TlFeH1xHP9sTEOwji5ljyPv2iN0/YdInkz+lF/4gEQSZ6hvLGGp+v3FUV66PU8dCusw9ysJhlJ40pvobz7MNdsiscDBx0zqiYUAdSR7XSgUQV7aOJH6dmnqwQA=
   file:
-    - release/terraform-provider-alks-darwin-amd64.tar.gz
-    - release/terraform-provider-alks-freebsd-386.tar.gz
-    - release/terraform-provider-alks-freebsd-amd64.tar.gz
-    - release/terraform-provider-alks-linux-386.tar.gz
-    - release/terraform-provider-alks-linux-amd64.tar.gz
-    - release/terraform-provider-alks-solaris-amd64.tar.gz
-    - release/terraform-provider-alks-windows-386.zip
-    - release/terraform-provider-alks-windows-amd64.zip
+    - release/terraform-provider-alks_$GIT_TAG_darwin_amd64.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_freebsd_386.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_freebsd_amd64.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_linux_386.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_linux_amd64.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_solaris_amd64.tar.gz
+    - release/terraform-provider-alks_$GIT_TAG_windows_386.zip
+    - release/terraform-provider-alks_$GIT_TAG_windows_amd64.zip
+    - release/terraform-provider-alks_$GIT_TAG_SHA256SUMS
+    - release/terraform-provider-alks_$GIT_TAG_SHA256SUMS.sig
```

Comment:
Why `$GIT_TAG` here over `$TRAVIS_TAG`?  I am also pretty afraid of the shell expansion also consuming the rest of the string.  Shouldn't this be `$(GIT_TAG)` or really rather `$(TRAVIS_TAG)` instead?

Reply:
I chose to manually extract the `$GIT_TAG` variable because I was under the impression that `$TRAVIS_TAG` was only being used within the `Makefile` file specifically. I think it is actually set as something like: `-X main.versionNumber=the_version_number`. If this is not the case and we can use it, then I would prefer to switch this variable out.

As far as the `$(TRAVIS_TAG)` versus `$TRAVIS_TAG` - this is because in the `Makefile` file they require variables to be in the first format, while the YAML file expects the "bash/shell" format for variables [as shown in this doc](https://docs.travis-ci.com/user/environment-variables/).

Reply:
Here's the reference for Travis variables that are available out of the box: https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

`$TRAVIS_TAG` is a special environment variable that Travis manages for builds triggered by a tag push.

We set a special linker flag (`-X ...`) during compilation so that we can dynamically reference the version information inside the executable, but it's not configuring anything specific to Travis.

`${}` is a bash/shell format, specifically a special escaping for when you've added a shell variable in the middle of a string that's also itself a valid shell variable.  Basically, shell gets weird when you do things like:
```sh
bash-3.2$ export MY_VAL=123
bash-3.2$ echo $MY_VAL
123
bash-3.2$ echo $MY_VAL_and_something

bash-3.2$ echo "$MY_VAL_and_something"

bash-3.2$ echo "${MY_VAL}_and_something"
123_and_something
bash-3.2$
```

Reply:
Okay great, updating to use the `TRAVIS_TAG` variable, and good note on why to use `${}` - to be quite honest, I've only ever used it because it always seemed to work in most scenarios. Good to know why! 😅